dask/dataframe/io/tests/test_parquet.py

```diff
@@ -228,8 +230,9 @@ def test_simple(tmpdir, write_engine, read_engine):
     df.set_index("a", inplace=True, drop=True)
     ddf = dd.from_pandas(df, npartitions=2)
     ddf.to_parquet(fn, engine=write_engine)
-    read_df = dd.read_parquet(fn, index=["a"], engine=read_engine)
-    assert_eq(ddf, read_df)
+    res = dd.read_parquet(fn, index=["a"], engine=read_engine)
+    sol = ddf.compute()
+    assert_eq(res, sol)
```

Comment:
Does this just need a `gather_statistics=True` to avoid the `compute()`? (same question for other usage of `sol` below)

Reply:
Or a `check_divisions=False`. The main change is that without a `_metadata` file `read_parquet` will result in a dataframe without divisions by default, and most of the tests are calling `assert_eq` on two dask dataframes (so divisions are compared). Since there are tests for explicitly checking divisions, in most cases checking divisions like this was accidental and unimportant, I used a mishmash of:

- Using a pandas dataframe for the intended solution
- Passing in `gather_statistics=True` to get a dataframe wit hknown divisions
- Passing in `check_divisions=False`

to resolve these failures.

Reply:
That makes sense - In my branch to deprecate `gather_statistics` (which I didn't submit yet, and will likely conflict with this PR's test chantes in many places), I did end up adding an explicit `calculate_divisions=True` in many places.

It is somewhat true that we are "accidentally" checking divisions, but my slight preference is to avoid computing before the `assert_eq` unless we really need to. I'd say `check_divisions=False` is a bit better, but I *may* end up rolling back some of these `sol` changes in the `gather_statistics` PR.